### DISCUSSION TITLE
Set global state from router on_update

### DISCUSSION THREAD

**Discussion by wackazong** [DioxusLabs/dioxus#2224] [Q&A] [answered]
How can I set a global state variable (using `consume_context::<Signal<T>>()`) in the router update callback? The update callback function is only `Fn` unfortunately, not `FnMut`.

**Comment by ealmloff** [comment, marked as answer]
Most methods to write to signals require `&mut self` to avoid overlapping reads and writes to the same signal, but you can opt-out of the extra checks with [write_unchecked](https://docs.rs/dioxus/latest/dioxus/prelude/trait.Writable.html#method.write_unchecked). If you only use that method inside the update function instead of `write`, it will implement `Fn`

**Reply by wackazong** [reply]
Ok, great, will do that.
